src/components/Competitions.jsx

```diff
@@ -1,9 +1,187 @@
-import React from 'react'
-
+import React, { useEffect, useState } from "react";
+import Filter from "./CompUtils/Filter";
+import CompetitionCard from "./CompUtils/CompetitionCard";
+import { compDetails } from "../constants/competition";
+import { logo } from "../assets";
 const Competitions = () => {
+  const [competitions, setCompetitions] = useState([]);
+  const [filterCompetitions, setFilterCompetitions] = useState([]);
+  const [activeFilter, setActiveFilter] = useState("All");
+
+  useEffect(() => {
+    setCompetitions(compDetails);
+    setFilterCompetitions(compDetails);
+  }, []);
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState("All Events");
+
+  const toggleDropdown = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const handleOptionSelection = (option) => {
+    setSelectedOption(option);
+    setTimeout(() => {
+      if (item === "All") {
+        setFilterCompetitions(competitions);
+      } else {
+        setFilterCompetitions(
+          competitions.filter((competitions) => competitions.category == item)
+        );
+      }
+    }, 500);
+    toggleDropdown();
+  };
+
+  const handleCompetitionFilter = (item) => {
+    setActiveFilter(item);
+
+    setTimeout(() => {
+      setSelectedOption(item);
+      if (item === "All") {
+        setFilterCompetitions(competitions);
+      } else {
+        setFilterCompetitions(
+          competitions.filter((competitions) => competitions.category == item)
+        );
+      }
+    }, 500);
+  };
+
   return (
-    <div>Competitions</div>
-  )
+    <section id="competitions">
+      <h1 className="flex-1 font-poppins font-semibold text-[45px] text-primary leading-[80px] justify-center">
+        Competitions
+      </h1>
+
+      {/* For coins */}
+      <div className="hidden md:block md:transition-all md:duration-100 md:flex flex-wrap md:flex-row md:space-x-9 md:justify-center md:items-center flex items-center justify-center">
+        <div class="w-auto text-white font-bold py-2 px-4 border-b-4 flex flex-row justify-evenly p-1 space-x-1">
+          {["All Events", "1", "2", "3", "4", "5", "6", "7"].map((item, index) => (
+            <button key={index}>
+              <img
+                src={logo}
+                alt=""
+                onClick={() => handleCompetitionFilter(item)}
```

Comment:
If you click "All events", `handleCompetitionFilter` will take "All Events" as the argument, while it should be taking "All" as the argument as in L94:L101.